src/structures/MessageAttachment.js

```diff
@@ -43,6 +43,23 @@ class MessageAttachment {
     return this;
   }
 
+  /**
+   * Sets whether this attachment is a spoiler
+   * @param {boolean} [spoiler=true] Whether the attachment should be marked as a spoiler
+   * @returns {MessageAttachment} This attachment
+   */
+  setSpoiler(spoiler = true) {
+    if (spoiler === this.spoiler) return this;
+    
+    if (!spoiler) {
+      while (this.spoiler) {
+        this.name.slice('SPOILER_'.length);
+      }
+    }
+    this.name = `SPOILER_${this.name}`;
```

Comment:
You either need an else here or a return statement above, otherwise this removes and then adds back the spoiler

Reply:
Alright